### PR TITLE
Scratch/categories

### PR DESCRIPTION
**Pull request by aditya9940**
categories creation

### CODE REVIEW

File: api/training/training.controller.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    }
+    //Create a new category
+    try {
+        const id = req.body.id;
+        const isExistingCategory = id && await db.models.Categories.findOne({ where: { id } })
+            .then((data) => {
+                return data;
+            });
+        let max_id = await db.models.CategoryDetail.max('id') || 0;
+        console.log(isExistingCategory, "MAX ID");
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:22]: remove this

File: api/training/training.controller.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        return;
+    }
+    //Create a new category
+    try {
+        const id = req.body.id;
+        const isExistingCategory = id && await db.models.Categories.findOne({ where: { id } })
+            .then((data) => {
+                return data;
+            });
+        let max_id = await db.models.CategoryDetail.max('id') || 0;
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:21]: This is for category detail id

File: api/training/training.controller.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+                language_id: req.body.language_id,
+                description: req.body.description,
+                category_id: req.body.id || categoryMaxId + 1,
+            }
+
+            if (!categoryDetailsData.language_id) {
+                return res.status(409).json({ error: 'language_id is required' });
+            }
+
+            console.log(categoryMaxId, max_id, "MAX ID");
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:107]: remove this console

File: api/training/training.controller.js [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+            });
+    }
+    catch (error) {
+        res.status(500).json({ message: error.message });
+    }
+};
+
+async function updateCategory(req, res) {
+    const id = Number(req.query.id);
+    console.log(id);
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:247]: remove this

File: api/training/training.controller.js [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+                    response.category = data;
+                    categoryDetailsData.category_id = data.id
+                    await db.models.CategoryDetail.create(categoryDetailsData)
+                        .then((data) => {
+                            response.CategoryDetail = data
+                            res.status(201).json(response)
+                        })
+                })
+                .catch((error) => {
+                    res.status(500).send({ message: error.message });
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:127]: can use use json here?

**Review by arijeet-mukherjee — APPROVED** [review]
LGTM

**Review by aditya9940 — COMMENTED** [review]
category creation

File: api/training/training.controller.js [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+            }
+
+            if (await isEntryPresentForLanguage(db.models.CategoryDetail, req.body.language_id, { category_id: req.body.id })) {
+                res.status(409).json({ message: "language entry exists for this Category" });
+                return;
+            }
+
+            const response = {}
+
+            await db.models.Categories.update(categoryData, { where: { id: req.body.id }, })
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:57]: If we are awaiting we should keep the creations logic in else block.

File: api/training/training.controller.js [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+            })
+        if (!record) {
+            res.status(409).json({ message: "Category or language dose not exist" });
+            return;
+        }
+
+
+        const categoryDetailsData = {
+            category_name: req.body.category_name,
+            description: req.body.description
```
Comment (arijeet-mukherjee) [inline api/training/training.controller.js:272]: Are we not updating gallery using it ?

Comment (aditya9940) [reply api/training/training.controller.js:272]: we are taking care of the gallery in the category table

File: api/training/training.controller.js (hunk 6 above)
Comment (aditya9940) [reply api/training/training.controller.js:57]: we'll take care of this in the subsequent PR

**Pull request merged**